gui/layers/_markers_layer.py

```diff
@@ -129,31 +156,31 @@ def size(self) -> Union[int, float, np.ndarray, list]:
         """float, ndarray: size of the marker symbol in px
         """
 
-        return self._size
+        return self._size_original
 
     @size.setter
     def size(self, size: Union[int, float, np.ndarray, list]) -> None:
 
-        if np.isscalar(size):
-            self._size = size
-
-            self.refresh()
-
-        elif isinstance(size, Iterable):
-            assert len(size) == len(self._coords), \
-             'If size is a list/array, must be the same length as '\
+        size_array = np.array(size)
+        if len(size_array.shape) == 2:
+            assert(size_array.shape == self._coords.shape), \
```

Comment:
OR, we could do the ambiguous thing, *except* in cases where it's ambiguous, and then we relax this check and make sure that we can broadcast. ie don't check that the shapes are identical, only that they can broadcast together.